adaptive_hci/datasets.py

```diff
@@ -29,6 +30,33 @@
 ]
 
 
+def get_episode_modes(episodes, n_samples_considered: int = 20):
+    modes = []
+    for ep in episodes:
+        unique_actions, counts = np.unique(ep.actions[:n_samples_considered], return_counts=True, axis=0)
+        main_action = unique_actions[np.argmax(counts)]
+        modes.append(main_action)
+
+    return modes
+
+
+def find_closest_episode(episodes, target_row):
+    ep_modes = get_episode_modes(episodes)
+
+    distances = np.linalg.norm(ep_modes - target_row, axis=1)
+    min_distance_indices = np.where(distances == distances.min())[0]
```

Comment:
`    distances = np.argmin(np.linalg.norm(ep_modes - target_row, axis=1))`

Reply:
argmin into random.choice crashed, I think only taking argmin is fine as well, shouldn't change much